katdal/h5datav3.py

```diff
@@ -169,7 +170,9 @@ class H5DataV3(DataSet):
     def __init__(self, filename, ref_ant='', time_offset=0.0, mode='r',
                  time_scale=None, time_origin=None, rotate_bls=False,
                  centre_freq=None, band=None, keepdims=False, **kwargs):
-        DataSet.__init__(self, filename, ref_ant, time_offset)
+        # The closest thing to a capture block ID is the Unix timestamp in the original filename
+        cbid = pathlib.Path(filename).stem
+        DataSet.__init__(self, cbid, ref_ant, time_offset, url=filename)
```

Comment:
With all this repetition it almost feels like we could have a H5Data super between DataSet and the H5DataV{n} classes. But since I do think we should have these classes deprecate at the next major version bump this is probably not worth doing...

Reply:
Agreed, there is a lot of historical overlap.